File: specex_pybind/src/mymath.cpp
// Licensed under a 3-clause BSD style license - see LICENSE.rst

#include <mymath.h>
#include <ext.h>

namespace spx = specex;

spx::MyMath::MyMath(int id){
  
  this->id = id;
  
}

int spx::MyMath::add(int a, int b){

  std::cout << "id in add = " << this->id << " " << std::endl;

  return ext_add(a, b);
}

File: specex_pybind/py/specex/mymath.py
# Licensed under a 3-clause BSD style license - see LICENSE.rst
# -*- coding: utf-8 -*-
"""
specex.mymath
=====================

Functions for testing trivial math in c++

"""
from __future__ import absolute_import, division, print_function

import numpy as np

import fitsio

from ._internal import (MYMATH_INT, MyMath)

def str_to_mymath_type(input):
    if input == "domymath":
        return MYMATH_INT
    else:
        raise ValueError("unknown target type '{}'".format(input))
    return None

def test_mymath(a,b):
    mm = MyMath(10)
    return mm.add(a,b)

